src/pages/studyView/geneSymbolSelection/GeneSelectionBox.tsx

```diff
@@ -0,0 +1,342 @@
+            };
+
+            let [entrezResults, hugoResults] = await Promise.all([getEntrezResults(), getHugoResults()]);
+            return {
+                found: [...entrezResults.found, ...hugoResults.found],
+                suggestions: [...entrezResults.suggestions, ...hugoResults.suggestions]
+            };
+        },
+        500
+    );
+
+    private readonly genes = remoteData({
+        invoke: () => this.invokeGenesLater(this.geneIds),
+        default: { found: [], suggestions: [] },
+        onResult: (genes) => {
+            let status = _.isUndefined(this.oql.error) && genes.suggestions.length === 0;
+            this.props.callback(this.oql.query, genes.found, this._geneQuery, status)
+        }
+    });
+
+    @computed get validationStatusMessage() {
+
+        if (this.oql.error)
+            return (
+                <span data-test='oqlError' className={styles.errorMessage}>
+                    {`Cannot validate gene symbols because of invalid OQL. ${
+                        this._geneQueryErrorDisplayStatus === DispalyStatus.UNFOCUSED
+                            ? "Please click 'Submit' to see location of error."
+                            : this.oql.error.message
+                        }`}
+                </span>
+            );
+
+        if (!this.oql.query.length)
+            return null;
+
+        if (this.genes.isError)
+            return (
+                <span className={styles.pendingMessage}>
+                    Unable to validate gene symbols.
+                </span>
+            );
+
+        if (this.genes.isPending && this.genes.result.suggestions.length == 0)
+            return (
+                <span className={styles.pendingMessage}>
+                    Validating gene symbols...
+                </span>
+            );
+
+        if (this.genes.result.suggestions.length)
+            return (
+                <div data-test='invalidGenes' style={{ display: 'flex' }}>
+                    <div className={styles.invalidBubble} title="Please edit the gene symbols.">
+                        <FontAwesome className={styles.icon} name='exclamation-circle' />
+                        <span>Invalid gene symbols.</span>
+                    </div>
+
+                    {this.genes.result.suggestions.map(this.renderSuggestion, this)}
+                </div>
+            );
+
+        if (this.props.location === GeneBoxType.STUDY_VIEW_PAGE) {
+            return null;
+        }
+        return (
+            <div className={styles.validBubble} title="You can now submit the list.">
+                <FontAwesome className={styles.icon} name='check-circle' />
+                <span>All gene symbols are valid.</span>
+            </div>
+        );
+    }
+
+    @action private replaceGene(oldSymbol: string, newSymbol: string) {
+        this._geneQuery = normalizeQuery(this._geneQuery.toUpperCase().replace(new RegExp(`\\b${oldSymbol.toUpperCase()}\\b`, 'g'), () => newSymbol.toUpperCase()));
```

Comment:
this is probably fine but again it's not really safe to do manual string processing on OQL (for example, what if there happens to be a gene with the same name as one of the OQL keywords... unlikely but not impossible). best to use OQL parser then modify that structured data